map_model/src/pathfind/v2.rs

```diff
@@ -119,11 +120,44 @@ impl PathV2 {
             }
         }
 
+        // The v2 path might immediately require a turn that's only available from some lanes. If
+        // the req.start lane can't make that turn, then producing the v1 path would fail. So let's
+        // allow starting from any lane on the same side of the road. Since petgraph can only start
+        // from a single node and since we want to prefer the originally requested lane anyway,
+        // create a virtual start node and connect it to all possible starting lanes.
+        let virtual_start_node = LaneID(map.lane_id_counter + 1);
```

Comment:
I considered a few other ideas instead of this virtual starting node:

- Refactoring our custom Dijkstra implementation used in `all_vehicle_costs_from`, which handles multiple sources. We'd have to extend that even further to handle initial weights for each starting node, though, and the hassle of refactoring doesn't seem worth it quite yet.
- Cutting over to some kind of lazy v2->v1 iterator, where we pick exact lanes only when required. That's still a bigger change I'd like to make, but it's complicated and kind of tangled up in all of the ideas floating around in #555. The virtual starting node approach here is a simple step forward with concrete results.